client/dashboard/data/index.ts

```diff
@@ -356,12 +327,9 @@ export const fetchEmail = ( id: string ): Promise< Email | undefined > => {
 };
 
 export const fetchUser = async (): Promise< User > => {
```

Comment:
In some instances (like `fetchSite` above) we remove the `async` keyword, but here we didn't. Any reason to not be consistent with our approach, or just an oversight?

Reply:
I updated the patch so that it doesn't modify any of these `async` keywords. It's true that most of these functions don't need to be async, they are merely normal promise-returning functions, but the change is off-topic for this PR.

I now limit the patch only to updating the actual `wp.req` calls to fix and improve their params.